Mathlib/RingTheory/Flat/Basic.lean

```diff
@@ -229,6 +240,146 @@ instance of_projective [h : Projective R M] : Flat R M := by
   cases h with
     | _ e he => exact of_retract R _ _ _ _ he
 
+open BigOperators in
+/--
+Define character module of `M` to be `M → ℚ ⧸ ℤ`
+If the character module of `M` is an injective module then `L ⊗ 𝟙 M` is injective for any linear map
+`L`.
+-/
+lemma rTensor_preserves_injective_linearMap_of_injective_characterModule
+    (h : Module.Injective R <| CharacterModule M) :
+    ∀ ⦃N N' : Type v⦄ [AddCommGroup N] [AddCommGroup N'] [Module R N] [Module R N']
+      (L : N →ₗ[R] N'), Function.Injective L → Function.Injective (L.rTensor M) := by
```

Comment:
Is there a reason you don't put all of this stuff before the colon?

Reply:
I think this way is more readable —— it is a complete sentence saying that the injectiveness of all linear maps  is preserved under tensoring with M

Reply:
Now it's an iff so you can no longer put anything before the colon :)